### PR TITLE
fix(Tearsheet): implement focus trapping within tearsheets

### PR DESCRIPTION
**Pull request by makafsal**
Contributes to #3499

This is to fix the focus trap issue in the Tearsheet component

#### What did you change?

1. File: `packages/ibm-products/src/components/Tearsheet/TearsheetShell.js`

Implement a 'keydown' event handler `handleKeyDown()` function that handles the 'Tab' key operations. The function checks for focusable elements inside the Tearsheet component and properly navigates the users to the focusable elements in the Tearsheet.

Change in the logic of focus claim function `claimFocus()`

2. File: `packages/ibm-products/src/components/CreateTearsheet/preview-components/MultiStepWithIntro.js`

Add `tabIndex` property conditionally to the `RadioTile` component

#### How did you test and verify your work?
Storybook tested in Chrome, Safari, Firefox, and Edge

### CODE REVIEW

**Review by amal-k-joy — APPROVED** [review]
Code looks fine and seems to be working.

**Comment by Ratheeshrajan** [comment]
@amal-k-joy @makafsal worth checking this utility https://github.com/carbon-design-system/ibm-products/blob/431e041534aa1151e2f3b622e111c97c4591cbfc/packages/ibm-products/src/global/js/utils/wrapFocus.js. This is used in SidePanel

**Comment by makafsal** [comment]
@Ratheeshrajan @elycheea @lee-chase 

I have tried this `wrapFocus()` utility function. Here are my observations from that:

1. When using the `usePortalTarget` hook to render the ComposedModal component, the `onBlur()` event is not firing, so I used the `onKeydown()` event to experiment with the _wrapFocus()_ function. It is working in a few cases, but it is not solving the issue in _stacked narrow tearsheet_ and _tearsheet with error state_.
2. In addition, focusing on dummy 'focus sentinal' elements may confuse users using screen readers.

However, I am trying to improve my current solution to make it more efficient, and reusable.

**Review by davidmenendez — DISMISSED** [review]

File: packages/ibm-products/src/global/js/hooks/useFocus.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright IBM Corp. 2020, 2023
```
Comment (elycheea) [inline packages/ibm-products/src/global/js/hooks/useFocus.js:2]:
```suggestion
 * Copyright IBM Corp. 2024, 2024
```

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**